src/main/java/org/opensearch/flowframework/transport/WorkflowRequest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.transport;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.Nullable;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.flowframework.model.Template;
+
+import java.io.IOException;
+
+/**
+ * Transport Request to create and provision a workflow
+ */
+public class WorkflowRequest extends ActionRequest {
+
+    /**
+     * The documentId of the workflow entry within the Global Context index
+     */
+    @Nullable
+    private String workflowId;
+    /**
+     * The use case template to index
+     */
+    @Nullable
+    private Template template;
+
+    /**
+     * Instantiates a new WorkflowRequest
+     * @param workflowId the documentId of the workflow
+     * @param template the use case template which describes the workflow
+     */
+    public WorkflowRequest(@Nullable String workflowId, @Nullable Template template) {
+        this.workflowId = workflowId;
+        this.template = template;
+    }
+
+    /**
+     * Instantiates a new Workflow request
+     * @param in The input stream to read from
+     * @throws IOException If the stream cannot be read properly
+     */
+    public WorkflowRequest(StreamInput in) throws IOException {
+        super(in);
+        this.workflowId = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.template = Template.parse(in.readString());
+        } else {
+            this.template = null;
+        }
+    }
+
+    /**
+     * Gets the workflow Id of the request
+     * @return the workflow Id
+     */
+    @Nullable
+    public String getWorkflowId() {
+        return this.workflowId;
+    }
+
+    /**
+     * Gets the use case template of the request
+     * @return the use case template
+     */
+    @Nullable
+    public Template getTemplate() {
+        return this.template;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeOptionalString(workflowId);
+        if (template != null) {
+            out.writeBoolean(true);
+            out.writeString(template.toJson());
```

Comment:
If you're just writing a string for the template here, you should use `writeOptionalString()` as above.  If you're serializing the whole template (making `Template` and its subclasses writeable) you probably want `writeOptionalWriteable()`.

Reply:
There is a case in which the template may be null, for instances in which a user may want to execute a template stored in an index, hence the null check prior to converting the template into a json string. I'll agree though, making the template class implement `Writable` seems like the better way to go about this. I'll make sure to add those modifications in the next commit.

Reply:
> There is a case in which the template may be null

So:
```
out.writeOptionalString(template == null ? null : template.toJson());
```
:-)